### PR TITLE
Use much simpler dockerfile for make test targets

### PR DESCRIPTION
**Pull request by mcgibbon**
## Purpose

Our docker building system previously had to support many features we no longer use, like compiling serialbox and the Fortran model. This PR adds a new Dockerfile which only includes needed features, and uses it as the image for docker testing.

After this PR, most docker build logic in the fv3gfs-physics, fv3core, and docker directories is obsolete and has been removed. Using identical images for fv3core and physics savepoint tests should make it easier to consolidate those tests.

## Infrastructure changes:

- Added top level `Dockerfile`
- Added top level `make build` target, which builds the `us.gcr.io/vcm-ml/pace` image
- Use `us.gcr.io/vcm-ml/pace` image for fv3core and physics image tests
- Deleted existing docker build logic, apart from postprocessing image

## Checklist
Before submitting this PR, please make sure:

- [x] You have followed the coding standards guidelines established at [Code Review Checklist](https://paper.dropbox.com/doc/Code-Review-Checklist--BD7zigBMAhMZAPkeNENeuU2UAg-IlsYffZgTwyKEylty7NhY).
- [x] Docstrings and type hints are added to new and updated routines, as appropriate
- [x] All relevant documentation has been updated or added (e.g. README, CONTRIBUTING docs)
- [x] For each public change and fix in `pace-util`, HISTORY has been updated
- [x] Unit tests are added or updated for non-stencil code changes

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -23,19 +23,18 @@ DEV ?=n
 CONTAINER_ENGINE ?=docker
 RUN_FLAGS ?=--rm
 CHECK_CHANGED_SCRIPT=$(CWD)/changed_from_main.py
-VOLUMES = -v $(CWD):/port_dev
-ifeq ($(DEV),y)
-	VOLUMES = -v $(CWD)/fv3core:/fv3core -v $(CWD)/fv3gfs-physics:/fv3gfs-physics -v $(CWD)/stencils:/stencils -v $(CWD)/dsl:/dsl -v $(CWD)/pace-util:/pace-util -v $(CWD)/test_data:/test_data -v $(CWD)/examples:/port_dev/examples -v $(CWD)/driver:/driver
-endif
```
Comment (elynnwu) [inline Makefile:29]: Why is this removed?

File: fv3core/Makefile [hunk 2]
```diff
@@ -38,8 +37,7 @@ PYTHON_FILES = $(shell git ls-files | grep -e 'py$$' | grep -v -e '__init__.py')
 PYTHON_INIT_FILES = $(shell git ls-files | grep '__init__.py')
 TEST_DATA_TARFILE=dat_files.tar.gz
 TEST_DATA_TARPATH=$(TEST_DATA_HOST)/$(TEST_DATA_TARFILE)
-CORE_TAR=$(SARUS_FV3CORE_IMAGE).tar
-MPIRUN_CALL ?=mpirun -np $(NUM_RANKS)
+MPIRUN_CALL ?=mpirun -np $(NUM_RANKS) --oversubscribe --mca btl_vader_single_copy_mechanism=none
```
Comment (elynnwu) [inline fv3core/Makefile:40]: Does this matter if we're running on gce vm vs. circle CI vm?

File: fv3core/Makefile [hunk 3]
```diff
@@ -21,14 +20,14 @@ TEST_DATA_HOST ?=$(TEST_DATA_ROOT)/$(FORTRAN_SERIALIZED_DATA_VERSION)/$(EXPERIME
...
 
 FV3=fv3core
-FV3_PATH ?=/$(FV3)
+FV3_PATH ?=/pace/$(FV3)
 
 ifeq ($(DEV),y)
 	VOLUMES += -v $(CWD):$(FV3_PATH) -v $(FV3UTIL_DIR):/usr/src/pace-util -v $(CWD)/external/stencils:/external/stencils  -v $(CWD)/external/dsl:/external/dsl
 endif
-CONTAINER_CMD?=$(CONTAINER_ENGINE) run $(RUN_FLAGS) $(VOLUMES) $(CUDA_FLAGS) $(FV3CORE_IMAGE)
+CONTAINER_CMD?=$(CONTAINER_ENGINE) run $(RUN_FLAGS) $(VOLUMES) $(CUDA_FLAGS) $(PACE_IMAGE)
```
Comment (elynnwu) [inline fv3core/Makefile:28]: Can we remove CUDA_FLAGS?

**Review by elynnwu — COMMENTED** [review]
Did some compile flag got changed? Looks like it's not passing on gce vm

File: Makefile (hunk 1 above)
Comment (mcgibbon) [reply Makefile:29]: Because instead the entire `pace` directory is mounted at `/pace`.

File: fv3core/Makefile (hunk 2 above)
Comment (mcgibbon) [reply fv3core/Makefile:40]: These are the commands for openmpi-based mpirun, it shouldn't matter. The only place we use mpich is Piz Daint, and there we pass this explicitly.

File: fv3core/Makefile (hunk 3 above)
Comment (mcgibbon) [reply fv3core/Makefile:28]: Yes.

Comment (mcgibbon) [reply fv3core/Makefile]: Removed.

**Review by elynnwu — APPROVED** [review]
lgtm!

File: fv3core/Makefile (hunk 2 above)
Comment (mcgibbon) [reply fv3core/Makefile]: Turns out this was an issue, looks like GCE uses mpich. I've added handling for this so the .jenkins-based workflow removes these flags if in the "virtualenv" setup.
